### PR TITLE
(chore) Serviceworker proxy

### PR DESCRIPTION
**Pull request by janjaap**
This PR adds a serviceworker that acts as a proxy that sits between the application and the API. It allows setting specific responses from endpoints that are used throughout the meldingen application.

To enable the proxy:

    PROXY=true HTTPS=true npm start

To configure URLs, see `sw-proxy-responses.js`. Each URL can contain wildcards (globs) and can contain response status codes and messages. The first entry in that file, for instance, disallows executing a `PATCH` request on a specific signal with id `1234`.

The proxy settings can be altered while the app is running, but you need to refresh the page to let the serviceworker consume those changes or even unregister the serviceworker before refreshing the page.

### CODE REVIEW

**Comment by aditudorache** [comment]
Looks nice! What is the intended use of this this sw-proxy?

**Comment by janjaap** [comment]
@aditudorache See https://medium.com/better-programming/how-to-run-a-proxy-server-inside-your-browser-8b96ea2ef1ea for reference

**Comment by janjaap** [comment]
@jpoppe @aditudorache Apologies for, again, a review of this PR. This will be the last. I've added documentation and updated the examples list of response objects. This PR is ready to be merged.

**Comment by jpoppe** [comment]
> @jpoppe @aditudorache Apologies for, again, a review of this PR. This will be the last. I've added documentation and updated the examples list of response objects. This PR is ready to be merged.

Top apologizing (I am getting paid for this ;))

File: server/cert/server.crt [hunk 1]
```diff
@@ -1,21 +1,17 @@
...
+gYSkYDHcIJbZZWkeTWklJmJPDs6BmUjaC3h2kB3ZjCoMY/WkG0UU0GLgcy5CVf/r
+lFTtkt1TW18P1fX0Pre8xTRclkQpwPbvMGhQl0WFJnbnRh6UtMj9aaVP3IPbrtSU
+DKjItMyJ4G6HLNcz5xXlc8VmvmnXueWMkZCA9oIUa6bF+xPHFLA9ofppjgmUUNII
+JeUZDjF5v3ss7OlbDHZAwveqltMCAwEAAaMYMBYwFAYDVR0RBA0wC4IJbG9jYWxo
+b3N0MA0GCSqGSIb3DQEBCwUAA4IBAQBMu8vgnbQCVXOOKL3BqCOwSDYy5YXRC5k/
+MKhG6NRpe9O/EiZOzWK7SW+tB6bhcdn1PaH2e8lMW0PTlBZW/XyNjPjLOyOBslWE
+hRUwjz0+cHTKXNgvbYoHE+zXnll4eZWhPaBF0ZBHA86TSfWHrCwlxjAWPd6IsabF
+ojnjCdokBZFCho0Z7e5fVaV+rPiVQ+49wadoT56CvLmmyIQpligFI0YGnUwckkkO
+u17F9hRp5nLelvtdOxKzRfj99gLSzIrtR1Y2Wxkfe21QYqIsi1WZhxnth+BjyHnn
+VPw9KrbunpaZ+ajd11n+24vwae7MvvzINHLbv7nJyHr8oKQKcAIS
```
Comment (jpoppe) [inline server/cert/server.crt:16]: While this certificate is only used for 'testing' purposes, I still have `weird` feelings by publishing it.

An alternative would to generate the certificates when they are not there when you start the proxy.

Comment (jpoppe) [reply server/cert/server.crt:16]: @janjaap what do you think about the proposal?

Comment (janjaap) [reply server/cert/server.crt:16]: Can you explain what feels 'weird' about having a self-signed certificate in this repo?

**Review by jpoppe — APPROVED** [review]

**Pull request merged**